common/src/main/java/com/wynntils/models/characterstate/CharacterStateModel.java

```diff
@@ -2,32 +2,41 @@
  * Copyright © Wynntils 2022.
  * This file is released under AGPLv3. See LICENSE for full license details.
  */
-package com.wynntils.models.characterstats;
+package com.wynntils.models.characterstate;
 
+import com.wynntils.core.WynntilsMod;
 import com.wynntils.core.components.Handlers;
 import com.wynntils.core.components.Model;
 import com.wynntils.core.components.Models;
-import com.wynntils.models.characterstats.actionbar.CoordinatesSegment;
-import com.wynntils.models.characterstats.actionbar.HealthSegment;
-import com.wynntils.models.characterstats.actionbar.ManaSegment;
-import com.wynntils.models.characterstats.actionbar.PowderSpecialSegment;
-import com.wynntils.models.characterstats.actionbar.SprintSegment;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.mc.event.TickEvent;
+import com.wynntils.models.characterstate.actionbar.CoordinatesSegment;
+import com.wynntils.models.characterstate.actionbar.HealthSegment;
+import com.wynntils.models.characterstate.actionbar.ManaSegment;
+import com.wynntils.models.characterstate.actionbar.PowderSpecialSegment;
+import com.wynntils.models.characterstate.actionbar.SprintSegment;
+import com.wynntils.models.characterstate.event.CharacterDeathEvent;
 import com.wynntils.models.elements.type.Powder;
 import com.wynntils.utils.MathUtils;
 import com.wynntils.utils.mc.McUtils;
 import com.wynntils.utils.type.CappedValue;
 import java.util.List;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
 
-public final class CharacterStatsModel extends Model {
+public final class CharacterStateModel extends Model {
     private final CoordinatesSegment coordinatesSegment = new CoordinatesSegment(this::centerSegmentCleared);
     private final HealthSegment healthSegment = new HealthSegment();
     private final ManaSegment manaSegment = new ManaSegment();
     private final PowderSpecialSegment powderSpecialSegment = new PowderSpecialSegment();
     private final SprintSegment sprintSegment = new SprintSegment();
 
-    public CharacterStatsModel(CombatXpModel combatXpModel) {
+    private static final String WYNN_DEATH_MESSAGE = "§r §4§lYou have died...";
+    private Vec3 lastLocation = new Vec3(0, 0, 0);
```

Comment:
Please use `Position` instead of `Vec3` unless you need to do vector operations.